pkg/webhook/validation.go

```diff
@@ -102,8 +102,11 @@ func (v *validator) validate(ctx context.Context, obj runtime.Object) (admission
 			sourceCount++
 			unionCount++
 
-			if len(configMap.Name) == 0 {
-				el = append(el, field.Invalid(path.Child("name"), configMap.Name, "source configMap name must be defined"))
+			if len(configMap.Name) == 0 && configMap.Selector == nil {
+				el = append(el, field.Invalid(path, "name: ' ', selector: nil", "must validate one and only one schema (oneOf): [name, selector]. Found none valid"))
+			}
+			if len(configMap.Name) > 0 && configMap.Selector != nil {
+				el = append(el, field.Invalid(path, fmt.Sprintf("name: %s, selector: {}", configMap.Name), "must validate one and only one schema (oneOf): [name, selector]. Found none valid"))
```

Comment:
```suggestion
				el = append(el, field.Invalid(path, fmt.Sprintf("name: %s, selector: {}", configMap.Name), "must validate one and only one schema (oneOf): [name, selector]. Found both set"))
```

I am not sure about the error message here, but it should indicate that we detected both set - which is invalid.

Reply:
I think `Found both set` is good. It makes sens to me at least.